### PR TITLE
[JENKINS-67627] Have expandable text boxes expand into multiple lines

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-67627](https://issues.jenkins-ci.org/browse/JENKINS-67627).

Alternative to #6201. This PR restores the previous behavior, expanding one line into multiple (while remaining safe from XSS).

Requesting @Wadeck as original author of the "regression", @NotMyFault as author of #6201, and @timja as participant in #6201.

### Proposed changelog entries

* Correctly render expandable text boxes into multiple lines (regression in 2.197 and 2.176.4).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/lib/form/ExpandableTextboxTest.java [hunk 1]
```diff
@@ -171,19 +174,37 @@ public void noXssUsingInputValue() throws Exception {
...
+
+        JenkinsRule.WebClient wc = j.createWebClient();
+        HtmlPage configurePage = wc.getPage(p, "configure");
+
+        HtmlInput input = configurePage.getElementByName("_.theField");
+        HtmlInput expandButton = (HtmlInput) input.getParentNode().getNextSibling().getFirstChild();
+        HtmlElementUtil.click(expandButton);
+        final DomElement textArea = configurePage.getElementByName("_.theField");
+        assertTrue(textArea instanceof HtmlTextArea);
+        assertEquals("foo\nbar\nbaz", ((HtmlTextArea) textArea).getText());
```
Comment (daniel-beck) [inline test/src/test/java/lib/form/ExpandableTextboxTest.java:192]: Without the fix:

```
[ERROR] Failures: 
[ERROR]   ExpandableTextboxTest.expandsIntoNewlines:192 expected:<foo[
bar
]baz> but was:<foo[bar]baz>

**Review by NotMyFault — APPROVED** [review]
This PR restores the previous behavior fine (and looks like the better approach 👀), expanding text boxes splits well into multiline args again.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by Wadeck — COMMENTED** [review]
Thanks for the correction, nice catch on the innerText instead of value!

**Pull request merged**